### ISSUE TITLE
[bug]: EnvInput causes restSession to be rendered twice

### ISSUE THREAD

**Issue by insub** [hoppscotch/hoppscotch#2363]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

when select another request, the code below will cause the restSession to be rendered twice：
<../components/smart/EnvInput.vue>
`
view.value?.dispatch({
  filter: false,
  changes: {
    from: 0,
    to: view.value.state.doc.length,
    insert: singleLinedText,
  },
})
`
will trigger  /newstore/RESTSession.ts -> setRESTEndpoint function.
PS: is the endpoint of the previous request was not "https://echo.hoppscotch.io"

### Steps to reproduce

1. open a request (make sure the endpoint was not "https://echo.hoppscotch.io")
2. open another request

### Environment

Release

### Version

Local

**Comment by AndrewBastin** [comment]
Are there any usage concerns (like a bug or a glitchy behavior) that may arise from this issue ?

**Comment by insub** [comment]
@AndrewBastin no, but it can be perceived in the user experience，slight delay
